# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/mlgpu/MLGDevice.cpp
# Commit: 1d9384906287
# Full Hash: 1d93849062870a8d492eeef2150ecea174cb6223
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2019-05-26 21:12:53
# Regressor Bug: 1550422
# File Overlap Count: 1
# Description:
#   Bug 1550422 - P15. Move Skip and Once gfxPrefs to StaticPrefs. r=jrmuizel
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D31259
# ==============================================================================

diff -r f5ebdcca640d -r 1d9384906287 gfx/layers/mlgpu/MLGDevice.cpp
--- a/gfx/layers/mlgpu/MLGDevice.cpp	Sun May 26 14:29:56 2019 +0000
+++ b/gfx/layers/mlgpu/MLGDevice.cpp	Sun May 26 14:30:14 2019 +0000
@@ -9,7 +9,7 @@
 #include "BufferCache.h"
 #include "ClearRegionHelper.h"
 #include "gfxConfig.h"
-#include "gfxPrefs.h"
+#include "mozilla/StaticPrefs.h"
 #include "gfxUtils.h"
 #include "LayersLogging.h"
 #include "ShaderDefinitionsMLGPU.h"
@@ -85,7 +85,7 @@
 
   // We allow this to be pref'd off for testing. Switching it off enables
   // Direct3D 11.0/Windows 7/OpenGL-style buffer code paths.
-  if (!gfxPrefs::AdvancedLayersEnableBufferSharing()) {
+  if (!StaticPrefs::AdvancedLayersEnableBufferSharing()) {
     gfxConfig::EnableFallback(Fallback::NO_CONSTANT_BUFFER_OFFSETTING,
                               "Disabled by pref");
     mCanUseConstantBufferOffsetBinding = false;
@@ -99,7 +99,7 @@
   // We allow this to be pref'd off for testing. Disabling it turns on
   // ID3D11DeviceContext1::ClearView support, which is present on
   // newer Windows 8+ drivers.
-  if (!gfxPrefs::AdvancedLayersEnableClearView()) {
+  if (!StaticPrefs::AdvancedLayersEnableClearView()) {
     mCanUseClearView = false;
   }
 
@@ -127,7 +127,7 @@
                 "Failed to allocate a shared shader buffer");
   }
 
-  if (gfxPrefs::AdvancedLayersEnableBufferCache()) {
+  if (StaticPrefs::AdvancedLayersEnableBufferCache()) {
     mConstantBufferCache = MakeUnique<BufferCache>(this);
   }
 